channeldb/invoices.go

```diff
@@ -135,90 +137,171 @@ const (
 	ampStateAmtPaidType     tlv.Type = 5
 )
 
-// AddInvoice inserts the targeted invoice into the database. If the invoice has
-// *any* payment hashes which already exists within the database, then the
-// insertion will be aborted and rejected due to the strict policy banning any
-// duplicate payment hashes. A side effect of this function is that it sets
-// AddIndex on newInvoice.
-func (d *DB) AddInvoice(newInvoice *invpkg.Invoice, paymentHash lntypes.Hash) (
-	uint64, error) {
+// WithTx is a helper method that allows callers to execute a function within a
```

Comment:
This is actually I doubt that I have because i am not sure if we can use the "component" that we are updating in the commit message instead of going "folder/package" based.